src/v/security/authorizer.h

```diff
@@ -113,16 +121,29 @@ class authorizer final {
           });
     }
 
-    void add_superuser(acl_principal principal) {
-        if (unlikely(seclog.is_shard_zero())) {
-            vlog(seclog.debug, "Adding superuser: {}", principal);
-        }
-        _superusers.emplace(std::move(principal));
-    }
-
 private:
     acl_store _store;
+
+    // The list of superusers is stored twice: once as a vector in the
+    // configuration subsystem, then again has a set here for fast lookups.
+    // The set is updated on changes via the config::binding.
     absl::flat_hash_set<acl_principal> _superusers;
+    config::binding<std::vector<ss::sstring>> _superusers_conf;
+    void update_superusers() {
+        for (const auto& username : _superusers_conf()) {
+            auto principal = acl_principal(principal_type::user, username);
+            if (!_superusers.contains(principal)) {
+                if (unlikely(seclog.is_shard_zero())) {
+                    vlog(
+                      seclog.info,
+                      "Registered superuser account: {}",
+                      principal);
+                }
+                _superusers.emplace(std::move(principal));
```

Comment:
was the removal of superusers intended to be supported? (at least it doesn't look like that'd happen here).

Reply:
eek!  Yes, it should be supported.  I wasn't paying enough attention when updating what was previously an add-only (at startup) path.

I've extended the unit test to cover revoking a superuser & fixed this function.